BDCSCli/Recipe.hs

```diff
@@ -43,6 +44,55 @@ import           Text.Read(readMaybe)
 import           Text.Toml(parseTomlDoc)
 
 import           BDCSCli.API.V0(Recipe(..), RecipeModule(..))
+import           BDCSCli.Utilities(maybeThrow)
+
+data GitError =
+    OpenRepoError
+  | CreateRepoError
+  | CreateBlobError
+  | CreateCommitError
+  | CreateBranchError
+  | BranchNameError
+  | WriteTreeError
+  | GetIndexError
+  | GetHeadError
+  | RefLookupError
+  | TreeBuilderError
+  | GetByNameError
+  | GetNameError
+  | GetTargetError
+  | GetTimeError
+  | GetTimeZoneError
+  | GetTreeError
+  | GetTreeIdError
+  | GetCommitterError
+  | GetMessageError
+  | GetParentsError
+  | LookupError
+  | LookupBlobError
+  | LookupBranchError
+  | LookupCommitError
+  | LookupTagError
+  | LookupTreeError
+  | LookupReferenceError
+  | RevparseError
+  | BuilderWriteError
+  | BuilderInsertError
+  | GetEntryIdError
+  | GetIdError
+  | GetRawBlobError
+  | GetTargetIdError
+  | NewOIdError
+  | NewOptionsError
+  | NewTimeValError
+  | NewTreeError
+  | NewSignatureError
+  | NewWalkerError
+  | OIdError
+  deriving (Eq, Show)
```

Comment:
I think this type would work better in its own file.  It could save having to move it out later should you end up needing to work around import cycles.